### PR TITLE
Allow setting optional checkout reference for SDK on CI workflows

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR addresses #780

### CODE REVIEW

**Review by dangeross — APPROVED** [review]
LGTM. I'm assuming an empty ref in the checkout defaults to main

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -28,6 +26,7 @@ jobs:
           repository: 'breez/breez-sdk'
           ssh-key: ${{secrets.REPO_SSH_KEY}}
           path: 'breez-sdk'
+          ref: ${{ inputs.ref }}
```
Comment (JssDWt) [inline .github/workflows/CI.yml:29]: There won't be an input on `push` or `pull request`, so perhaps add a fallback to the github sha to make sure the field is always set.

```suggestion
          ref: ${{ inputs.ref || github.sha }}
```

File: .github/workflows/build-android.yml [hunk 2]
```diff
@@ -57,6 +53,7 @@ jobs:
           repository: 'breez/breez-sdk'
           ssh-key: ${{secrets.REPO_SSH_KEY}}
           path: 'breez-sdk'
+          ref: ${{ inputs.ref }}
```
Comment (JssDWt) [inline .github/workflows/build-android.yml:56]: I would suggest the same here

```suggestion
          ref: ${{ inputs.ref || github.sha }}
```

File: .github/workflows/build-ios.yml [hunk 3]
```diff
@@ -75,6 +71,7 @@ jobs:
           repository: 'breez/breez-sdk'
           ssh-key: ${{secrets.REPO_SSH_KEY}}
           path: 'breez-sdk'
+          ref: ${{ inputs.ref }}
```
Comment (JssDWt) [inline .github/workflows/build-ios.yml:74]:
```suggestion
          ref: ${{ inputs.ref || github.sha }}
```

**Review by JssDWt — COMMENTED** [review]
some comments about fallback if the CI isn't run through workflow_dispatch

**Comment by erdemyerebasmaz** [comment]
> LGTM. I'm assuming an empty ref in the checkout defaults to main

@dangeross It is according to docs
```
# The branch, tag or SHA to checkout. When checking out the repository that
# triggered a workflow, this defaults to the reference or SHA for that event.
# Otherwise, uses the default branch.
ref: ''
```

Since we're checking out a different repository, not the one that triggered the workflow, it should use the default branch.

**Comment by JssDWt** [comment]
> LGTM. I'm assuming an empty ref in the checkout defaults to main

Maybe we can make this explicit, just to be sure?

```
${{ inputs.ref || 'main' }}
```

**Comment by erdemyerebasmaz** [comment]
> > LGTM. I'm assuming an empty ref in the checkout defaults to main
> 
> Maybe we can make this explicit, just to be sure?
> 
> ```
> ${{ inputs.ref || 'main' }}
> ```

Sure, I'll explicitly set the default value to `main`

**Pull request merged**